client/my-sites/email/email-providers-stacked-comparison/provider-cards/google-workspace-card.tsx

```diff
@@ -78,23 +78,36 @@ const googleWorkspaceCardInformation: ProviderCard = {
 };
 
 const GoogleWorkspaceCard = ( {
+	cartDomainName,
 	comparisonContext,
-	currencyCode = '',
 	detailsExpanded,
-	domain,
-	gSuiteProductMonthly,
-	gSuiteProductYearly,
-	hasCartDomain,
 	intervalLength,
 	onExpandedChange,
 	selectedDomainName,
-	selectedSite,
-	shoppingCartManager,
 	source,
 }: EmailProvidersStackedCardProps ): ReactElement => {
 	const googleWorkspace: ProviderCard = { ...googleWorkspaceCardInformation };
 	googleWorkspace.detailsExpanded = detailsExpanded;
 
+	const hasCartDomain = Boolean( cartDomainName );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const selectedSite = useSelector( getSelectedSite );
+	const domains = useSelector( ( state ) => getDomainsBySiteId( state, selectedSite?.ID ) );
+	const domain = getSelectedDomain( {
+		domains,
+		selectedDomainName: selectedDomainName,
+	} );
+
+	const cartKey = useCartKey();
+	const shoppingCartManager = useShoppingCart( cartKey );
+
+	const gSuiteProductYearly = useSelector( ( state ) =>
+		getProductBySlug( state, GOOGLE_WORKSPACE_BUSINESS_STARTER_YEARLY )
+	);
+	const gSuiteProductMonthly = useSelector( ( state ) =>
+		getProductBySlug( state, GOOGLE_WORKSPACE_BUSINESS_STARTER_MONTHLY )
+	);
+
```

Comment:
Nit: We don't need to define the two types of product here. We could simply use the product slug and interval to build a product